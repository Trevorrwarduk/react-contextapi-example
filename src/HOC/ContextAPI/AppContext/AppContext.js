import React, { Component } from "react";

export const AppContext = React.createContext();

export const AppContextConsumer = AppContext.Consumer;

class AppContextProvider extends Component {
  titles = {
    homePage: [
      "Home Page",
      "Panel One Selected",
      "Panel Two Selected",
      "Panel Three Selected"
    ],
    controlPanel: [
      "Control Panel",
      "Panel One Control!",
      "Panel Two Control!",
      "Panel Three Control!"
    ],
    panelOne: [
      "Panel One",
      "Panel One Displaying itself",
      "Panel Two On Panel One!",
      "Panel Three On Panel One!"
    ],
    panelTwo: [
      "Panel Two",
      "Panel One On Panel Two",
      "Panel Two Displaying itself",
      "Panel Three On Panel Two"
    ],
    panelThree: [
      "Panel Three",
      "Panel One On Panel Three",
      "Panel Two On Panel Three",
      "Panel Three Displaying itself"
    ]
  };
  state = {
    homePageTitle: this.titles.homePage[0],
    controlPaneltitle: this.titles.controlPanel[0],
    panelOneTitle: this.titles.panelOne[0],
    panelTwoTitle: this.titles.panelTwo[0],
    panelThreeTitle: this.titles.panelThree[0]
  };

  updateTitlesFunc = inId => {
    this.setState({
      homePageTitle: this.titles.homePage[inId],
      controlPaneltitle: this.titles.controlPanel[inId],
      panelOneTitle: this.titles.panelOne[inId],
      panelTwoTitle: this.titles.panelTwo[inId],
      panelThreeTitle: this.titles.panelThree[inId]
    });
  };

  render() {
    return (
      <AppContext.Provider
        value={{
          state: this.state,
          updateTitles: this.updateTitlesFunc
        }}
      >
        {this.props.children}
      </AppContext.Provider>
    );
  }
}

export default AppContextProvider;
