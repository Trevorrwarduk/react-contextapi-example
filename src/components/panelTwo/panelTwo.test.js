import React from "react";

import { create } from "react-test-renderer";

import { MdAddCircle } from "react-icons/md";
import {
  PanelContainer,
  ControlIcon,
  StyledRow,
  StyledCol
} from "../sharedStyles/sharedStyles";

import { AppContext } from "../../HOC/ContextAPI/AppContext/AppContext";
import context from "../../__mocks__/AppContextMock";

import PanelTwo from "./panelTwo";

describe("<PanelOne> ", () => {
  let renderedComponent;
  let testInstance;

  beforeEach(() => {
    renderedComponent = create(
      <AppContext.Provider value={context}>
        <PanelTwo />
      </AppContext.Provider>
    );
    testInstance = renderedComponent.root;
  });

  it("should test the Panel Snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it("should check the correct style elements are displayed", () => {
    const panelChk = testInstance.findAllByType(PanelContainer);
    const styRowChk = testInstance.findAllByType(StyledRow);
    const styColChk = testInstance.findAllByType(StyledCol);
    const iconChk = testInstance.findAllByType(MdAddCircle);

    expect(panelChk).toHaveLength(1);
    expect(styRowChk).toHaveLength(1);
    expect(styColChk).toHaveLength(2);
    expect(iconChk).toHaveLength(1);

    // Check the text displayed
    expect(styColChk[1].children[0].props.children).toEqual("Panel Two");
  });

  it("should check the correct icon is displayed and the function is called", () => {
    const contIconChk = testInstance.findAllByType(ControlIcon);

    expect(contIconChk).toHaveLength(1);

    contIconChk[0].props.onClick();

    expect(context.updateTitles).toHaveBeenCalled();
    expect(context.updateTitles).toHaveBeenCalledWith(2);
  });
});
