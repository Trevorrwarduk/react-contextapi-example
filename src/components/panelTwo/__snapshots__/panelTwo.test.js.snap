// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelOne>  should test the Panel Snapshot 1`] = `
<div
  className="sc-cvbbAY kSlAKp"
>
  <div
    className="sui-row sc-dxgOiQ sc-iAyFgw fVRdiY"
  >
    <div
      className="sui-col sc-EHOje sc-hSdWYo cwHTXc"
    >
      <button
        className="sc-brqgnP hprPVB"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
      </button>
    </div>
    <div
      className="sui-col sc-EHOje sc-hSdWYo cwHTXc"
    >
      Panel Two
    </div>
  </div>
</div>
`;
